tests/test_user.py

```diff
@@ -458,3 +461,504 @@ def test_user_settings_edit_dev_app(self, driver, session, fake):
+                'osf.full_write',
+                'osf.nodes.metadata_write',
+                'osf.nodes.access_write',
+                'osf.nodes.data_write',
+                'osf.users.profile_write',
+                'osf.nodes.full_read',
+                'osf.nodes.metadata_read',
+                'osf.nodes.access_read',
+                'osf.nodes.data_read',
+                'osf.users.profile_read',
+                'osf.users.email_read',
+            ]
+            self.verify_scope_checkboxes(
+                driver, edit_page, unselected_scopes, checked=False
+            )
+
+            # Make some Edits - change the token name and change permissions from
+            # osf.full_read to osf.full_write
+            new_token_name = token_name + ' edited'
+            edit_page.scroll_into_view(edit_page.token_name_input.element)
+            edit_page.token_name_input.clear()
+            edit_page.token_name_input.send_keys(new_token_name)
+            edit_page.scroll_into_view(edit_page.osf_full_read_checkbox.element)
+            edit_page.osf_full_read_checkbox.click()
+            edit_page.scroll_into_view(edit_page.osf_full_write_checkbox.element)
+            edit_page.osf_full_write_checkbox.click()
+
+            # Click the Save button
+            edit_page.scroll_into_view(edit_page.save_button.element)
+            edit_page.save_button.click()
+
+            # Should end up back on PAT list page with new token name listed
+            pat_page = user.PersonalAccessTokenPage(driver, verify=True)
+            pat_page.loading_indicator.here_then_gone()
+            pat_card = pat_page.get_pat_card_by_name(new_token_name)
+            assert pat_card
+
+            # Now click the PAT name link to go back to the Edit PAT page and verify
+            # the data changes
+            pat_link = pat_card.find_element_by_css_selector('a')
+            pat_link.click()
+            edit_page = user.EditPersonalAccessTokenPage(driver, verify=True)
+            assert edit_page.token_name_input.get_attribute('value') == new_token_name
+
+            # Use helper function to verify scopes
+            selected_scopes = ['osf.full_write']
+            self.verify_scope_checkboxes(
+                driver, edit_page, selected_scopes, checked=True
+            )
+            unselected_scopes = [
+                'osf.nodes.full_write',
+                'osf.full_read',
+                'osf.nodes.metadata_write',
+                'osf.nodes.access_write',
+                'osf.nodes.data_write',
+                'osf.users.profile_write',
+                'osf.nodes.full_read',
+                'osf.nodes.metadata_read',
+                'osf.nodes.access_read',
+                'osf.nodes.data_read',
+                'osf.users.profile_read',
+                'osf.users.email_read',
+            ]
+            self.verify_scope_checkboxes(
+                driver, edit_page, unselected_scopes, checked=False
+            )
```

Comment:
Golfing a bit here, but i think this'll make it a little shorter and more readable:

1. Create a `list` with all possible scopes in it: `all_scopes = ['osf.$foo', 'osf.$bar', ...etc.]`
2. Make the third arg to `verify_scope_checkboxes` a `dict` rather than a list and drop the fourth arg. The key of the `dict` is a scope name, the value is a boolean for whether it's checked or not. Initialize all values to the most common state, then manually override those that differ. The above would become:

```
scope_perms = {p:False for p in all_scopes}
scope_perms['osf.full_write'] = True
self.verify_scope_checkboxes(driver, edit_page, scope_perms)
```

3. Update `verify_scope_checkboxes`: rename `scope_list` to `scope_perms`, then update line 490 to `for scope, perm in scope_perms.items():`, finally update the assert to `assert checkbox.is_selected() == perm`

The line savings will be most dramatic for tests where most of the scopes' permissions are the same (like this one).  The savings will be less apparent when there are equal numbers of both, but it should still be a slight net win.  It'll also save you an extra call to `verify_scope_checkboxes` for each test.

Reply:
Ok. I think I got it now. Thanks!